compiler/src/iree/compiler/Codegen/Common/TransformDialectJitterPass.cpp

```diff
@@ -391,31 +436,115 @@ class TransformDialectJitterPass
     linalg::registerTransformDialectExtension(registry);
   }
 
-  TransformDialectJitterPass() = default;
+  LogicalResult buildStrategy(ImplicitLocOpBuilder ib, Value variantH,
+                              StringRef target, StringRef strategyName,
+                              ArrayRef<int64_t> workgroupSize) {
+    // TODO: this is very brittle.
+    if (target == "cuda") {
+      if (strategyName == "reduction_strategy")
+        return buildReductionCudaStrategy(ib, variantH, workgroupSize);
+    } else if (target == "llvm-cpu") {
+      if (strategyName == "reduction_strategy")
+        return buildReductionCpuStrategy(ib, variantH, workgroupSize);
+    }
+    return getOperation()->emitOpError("Unknown strategy: ");
```

Comment:
I suppose this also needs to print the strategy name in the error message.

Also nit: use `emitError` rather than `emitOpError` when the error is not related to the properties of the given op. Practically, the latter will prefix the message with "custom op foo.bar", which makes it read weird.